src/app/components1/list-product-detail/list-product-detail.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, OnInit } from '@angular/core';
+import { ProductService } from '../product.service';
+import Product from '../Types/Product';
+import { AuthentificationService } from '../authentification.service';
+@Component({
+  selector: 'app-list-product-detail',
+  templateUrl: './list-product-detail.component.html',
+  styleUrls: ['./list-product-detail.component.scss'],
+})
+export class ListProductDetailComponent implements OnInit {
+  products: Product[] = [];
+  canEditProducts: boolean = this.authService.userHasRole('admin');
+  constructor(private productService: ProductService,private authService: AuthentificationService) {}
+
+  ngOnInit(): void {
+    this.get();
+  }
+
+  get(): void {
+    this.productService.getAllProducts().subscribe((data) => {
```

Comment:
**suggestion:** take care to avoid memory leaks in your code by unsubscribing from observables. https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions/